app/frontend/src/service/storage.ts

```diff
@@ -1,149 +1,314 @@
-        let db = openRequest.result;
-        if (!db.objectStoreNames.contains(storage.objectStore_name)) {
-            db.createObjectStore(storage.objectStore_name, { keyPath: "id" });
-        }
-    };
-    openRequest.onerror = function () {
-        console.error("Error", openRequest.error);
-    };
+export function popLastMessageInDB(storage: indexedDBStorage, id: number) {
+    let openRequest = indexedDB.open(storage.db_name, storage.db_version);
+    openRequest.onupgradeneeded = () => onUpgrade(openRequest, storage);
+    openRequest.onerror = () => onError(openRequest);
     openRequest.onsuccess = function () {
-        let db = openRequest.result;
-        let transaction = db.transaction(storage.objectStore_name, "readwrite");
-        let chat = transaction.objectStore(storage.objectStore_name);
-        let stored = chat.get(ID);
+        let chat = openRequest.result.transaction(storage.objectStore_name, "readwrite").objectStore(storage.objectStore_name);
+        let stored = chat.get(id);
         stored.onsuccess = function () {
-            let deleted = chat.delete(ID);
+            let deleted = chat.delete(id);
             deleted.onsuccess = function () {
                 if (stored.result) {
-                    stored.result.Answers.pop();
-                    let put = chat.put({ Answers: stored.result.Answers, id: ID });
-                    put.onerror = function () {
-                        console.error("Error", put.error);
-                    };
+                    stored.result.Data.Answers.pop();
+                    let put = chat.put(stored.result);
+                    put.onerror = () => onError(put);
                 }
             };
-            deleted.onerror = function () {
-                console.error("Error", deleted.error);
+            deleted.onerror = () => onError(deleted);
+        };
+        stored.onerror = () => onError(stored);
+    };
+}
+
+export function getHighestKeyInDB(storage: indexedDBStorage): Promise<number> {
+    return new Promise((resolve, reject) => {
+        let openRequest = indexedDB.open(storage.db_name, storage.db_version);
+        openRequest.onupgradeneeded = () => onUpgrade(openRequest, storage);
+        openRequest.onerror = () => onError(openRequest);
+        openRequest.onsuccess = function () {
+            let keys = openRequest.result.transaction(storage.objectStore_name, "readonly").objectStore(storage.objectStore_name).getAllKeys();
+            keys.onsuccess = function () {
+                let highestKey = Math.max(...keys.result.map(Number), 0);
+                resolve(highestKey);
             };
+            keys.onerror = () => reject(keys.error);
         };
-        stored.onerror = function () {
-            console.error("Error", stored.error);
+    });
+}
+
+export function getZeroChat(storage: indexedDBStorage): Promise<number | undefined> {
+    return new Promise((resolve, reject) => {
```

Comment:
Was heißt, getZeroChat, "Kurzen Kommentar" der die Methode beschreibt. Sonst kennt sich da niemand mehr in einem Monat aus.